mdm/common/src/main/java/com/gentics/mesh/core/jobs/MailJobProcessor.java

```diff
@@ -0,0 +1,248 @@
+import com.gentics.mesh.core.data.binary.HibBinary;
+import com.gentics.mesh.core.data.branch.HibBranch;
+import com.gentics.mesh.core.data.dao.BinaryDao;
+import com.gentics.mesh.core.data.dao.JobDao;
+import com.gentics.mesh.core.data.dao.NodeDao;
+import com.gentics.mesh.core.data.dao.PersistingJobDao;
+import com.gentics.mesh.core.data.job.HibJob;
+import com.gentics.mesh.core.data.job.HibMailJob;
+import com.gentics.mesh.core.data.node.HibNode;
+import com.gentics.mesh.core.data.node.field.HibBinaryField;
+import com.gentics.mesh.core.data.project.HibProject;
+import com.gentics.mesh.core.db.Database;
+import com.gentics.mesh.core.db.Tx;
+import com.gentics.mesh.core.endpoint.migration.MigrationStatusHandler;
+import com.gentics.mesh.core.migration.impl.MigrationStatusHandlerImpl;
+import com.gentics.mesh.core.rest.admin.mail.MailAttachmentsRequest;
+import com.gentics.mesh.core.rest.admin.mail.MailSendingRequest;
+import com.gentics.mesh.core.rest.event.node.BranchMigrationCause;
+import com.gentics.mesh.core.rest.event.node.MailSendingCause;
+import com.gentics.mesh.core.rest.schema.BinaryFieldSchema;
+import com.gentics.mesh.core.rest.schema.FieldSchema;
+import com.gentics.mesh.etc.config.MailOptions;
+import com.gentics.mesh.etc.config.MeshOptions;
+import com.gentics.mesh.json.JsonUtil;
+import com.gentics.mesh.parameter.impl.NodeParametersImpl;
+import com.gentics.mesh.parameter.impl.VersioningParametersImpl;
+import com.gentics.mesh.util.RxUtil;
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.reactivex.SingleSource;
+import io.reactivex.functions.Consumer;
+import io.reactivex.functions.Function;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.mail.MailAttachment;
+import io.vertx.ext.mail.MailConfig;
+import io.vertx.ext.mail.MailMessage;
+import io.vertx.ext.mail.StartTLSOptions;
+import io.vertx.reactivex.core.Vertx;
+import io.vertx.reactivex.ext.mail.MailClient;
+
+/**
+ * This class is responsible for starting a mail sending from a job
+ */
+public class MailJobProcessor implements SingleJobProcessor {
+
+	public static final Logger log = LoggerFactory.getLogger(MailJobProcessor.class);
+	private final Database db;
+	private final PersistingJobDao jobDao;
+	private final Vertx vertx;
+	private final MeshOptions meshOptions;
+
+	@Inject
+	public MailJobProcessor(Database db, JobDao jobDao, Vertx vertx, MeshOptions meshOptions) {
+		this.db = db;
+		this.jobDao = (PersistingJobDao) jobDao;
+		this.vertx = vertx;
+		this.meshOptions = meshOptions;
+
+	}
+
+	@Override
+	public Completable process(HibJob job) {
+		HibMailJob mailJob = (HibMailJob) job;
+		MailConfig config = getConfig(meshOptions.getMailOptions());
+		MailClient mailClient = MailClient.create(vertx, config);
+		String mail = db.tx(mailJob::getMail);
+
+		MailSendingRequest request = JsonUtil.readValue(mail, MailSendingRequest.class);
```

Comment:
These approach is not that much useful. Better to send/store the mail parts distinctly, as done in the sending context. This will allow the client to fetch/filter/reuse the jobs, if required, and gives better control over sending.

Reply:
The problem is that I have also data that cannot be send in primitive data like Lists of Strings for example